### DISCUSSION TITLE
uIP limiting number of routes

### DISCUSSION THREAD

**Discussion by mohamx** [contiki-ng/contiki-ng#1637] [Q&A]
I would like to limit the number of routes that join the network to 15 even if there are more and I have a problem.
I decreased “UIP_CONF_DS6_ROUTE_NBU” in the "config-uip-public.h" file to 15 instead of 200 routes but what happened is:
1-	More than 15 nodes joined the network but I want only 15 nodes to join.
2-	The RPL table in the root coordinator only stores 15 routes and if there is an incoming DAO message from a new node, 
the RPL removes the old one and stores the new one.
How can I limit the network into a specified number and in the same time the other nodes search for other root coordinators to join.

**Comment by atiselsts** [comment]
It's not possible to limit the number of nodes that have joined the RPL network. RPL is a distributed protocol, a node does not have to ask permission to a central controller in order to join the network. The other network nodes do not have information about how many nodes have already joined the network.

The only situation when you can reliably limit this to some extent is when you have a star network. Then it's possible to limit the number of nodes that are in the gateway's routing table. But realistically, even in this situation the other nodes will still be able to receive messages from the gateway and send messages to it, so "joining the network" is somewhat a misleading term.
